### ISSUE TITLE
Rebuild a PipelineRun leads to an error in the Dashboard

### ISSUE THREAD

**Issue by maxveit** [tektoncd/dashboard#539]
# Expected Behavior
"Rebuild" of an PipelineRun leads to the execution of a rebuild.

# Actual Behavior
"Rebuild" of an PipelineRun leads to an error:
```
{
   "level":"error",
   "msg":"an error occurred rebuilding the PipelineRun java-microprofile-build-deploy-pipeline-run-1569231654541 in namespace kabanero: Internal error occurred: admission webhook \"webhook.tekton.dev\" denied the request: mutation failed: Invalid resource name: length must be no more than 63 characters: metadata.name"
}
```

# Steps to Reproduce the Problem
1. Fresh install of Kabanero
2. Create a Git Resource, I used the following:
```
apiVersion: tekton.dev/v1alpha1
kind: PipelineResource
metadata:
  name: stocktrader-trader-git
  namespace: kabanero
spec:
  type: git
  params:
    - name: url
      value: https://github.com/IBMStockTrader/trader.git
    - name: revision
      value: master
```
3. Create a Docker Resource, I used the following (ImageStream created beforehand):
```
apiVersion: tekton.dev/v1alpha1
kind: PipelineResource
metadata:
  name: stocktrader-trader-image
  namespace: kabanero
spec:
  type: image
  params:
    - name: url
      value: docker-registry.default.svc:5000/kabanero/stocktrader-trader
```
4. Create a PipelineRun with the above parameters and the microprofile pipeline (no ServiceAccount)
5. Click Rebuild

# Additional Info
n/a

**Comment by a-roberts** [comment]
Hmmm, we add -r-*generated string, five chars* for rebuilds, some interesting info pertinent to this

1) the 63 char limit comes from Tekton Pipelines itself
2) based on the timestamp being added to your run I see this has been made through a webhook - we could:
    * shorten the timestamp value made with a webhook
    * trim your pipeline name or step names owing to this Tekton restriction
    * decide on a new naming convention for rebuilds (e.g. looking for -r-*x* and just generating a new x

I'd like to defer getting this done for the upcoming 0.2.0 release unfortunately as we're swamped with other bugs, but @mnuttall FYI

**Comment by a-roberts** [comment]
I'm going to see if the fix I put in for https://github.com/tektoncd/dashboard/issues/567 resolves this as I modified the behaviour to use generateName

**Comment by a-roberts** [comment]
@maxveit hmm can you share the microprofile pipeline please, and let us know which version of the Tekton Dashboard/Webhooks Extension you used? FYI I published 0.2.0 of both y'day and maybe this helps you

**Comment by a-roberts** [comment]
This shouldn't happen now with the new generateName before delivered, if it's still seen with the 0.2.0 code or above please reopen this and thanks for the report
